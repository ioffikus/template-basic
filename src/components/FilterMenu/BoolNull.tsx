import React from 'react';
import _ from 'lodash';
import { Radio } from 'antd';

interface IProps {
  ui: Alicanto.Models.UI;
  value: boolean;
  fieldSlug: string;
  valueType: string;
  filterType: string;
  onChange: (value: boolean, fieldSlug: string, valueType: string, filterType: string) => void;
}

interface IState {}

class BoolNull extends React.PureComponent<IProps, IState> {
  handleChange = (event: React.SyntheticEvent<any>) => {
    const value = JSON.parse(_.get(event.target, 'value'));
    this.props.onChange(value, this.props.fieldSlug, this.props.valueType, this.props.filterType);
  };

  render() {
    const stateUI = this.props.ui.common.filters.inStock.state;
    const value = _.get(this.props, 'value', null);
    return (
      <div className="root">
        <Radio.Group value={'' + value} size="large" onChange={this.handleChange}>
          <Radio.Button value="true">{stateUI.trueState.i18n.defaultText}</Radio.Button>
          <Radio.Button value="false">{stateUI.falseState.i18n.defaultText}</Radio.Button>
        </Radio.Group>
      </div>
    );
  }
}

export default BoolNull;
