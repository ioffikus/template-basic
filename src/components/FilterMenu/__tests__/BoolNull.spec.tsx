import React from 'react';
import { shallow } from 'enzyme';
import { Radio } from 'antd';
import uiStateMock from 'src/store/__mocks__/uiStore';
import filterStateMock from 'src/store/__mocks__/filterStore';
import BoolNull from '../BoolNull';

describe(null, () => {
  it('render() correct with initialState', () => {
    const boolNullMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = boolNullMock;
    const value: boolean = null;
    const onChange = jest.fn();
    const wrapper = shallow(
      <BoolNull
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        onChange={onChange}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render() correct with value=true', () => {
    const boolNullMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = boolNullMock;
    const value = true;
    const onChange = jest.fn();
    const wrapper = shallow(
      <BoolNull
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        onChange={onChange}
      />,
    );
    const radioGroupInstance = wrapper
      .find(Radio.Group)
      .dive()
      .instance() as Radio;
    expect(radioGroupInstance.props.value).toEqual('' + value);
    expect(wrapper).toMatchSnapshot();
  });
  it('BoolNull.onChange()', () => {
    const boolNullMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = boolNullMock;
    const value: boolean = null;
    const newValue = true;
    const onChange = jest.fn();
    const wrapper = shallow(
      <BoolNull
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        onChange={onChange}
      />,
    );
    const radioGroupInstance = wrapper
      .find(Radio.Group)
      .dive()
      .instance() as Radio;
    radioGroupInstance.props.onChange({ target: { value: newValue } } as any);
    expect(onChange).toBeCalledWith(newValue, fieldSlug, valueType, filterType);
  });
});
