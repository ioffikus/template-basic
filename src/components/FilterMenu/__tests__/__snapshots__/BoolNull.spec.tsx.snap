// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correct with initialState 1`] = `
<div
  className="root"
>
  <RadioGroup
    disabled={false}
    onChange={[Function]}
    size="large"
    value="null"
  >
    <RadioButton
      prefixCls="ant-radio-button"
      value="true"
    >
      Yes
    </RadioButton>
    <RadioButton
      prefixCls="ant-radio-button"
      value="false"
    >
      No
    </RadioButton>
  </RadioGroup>
</div>
`;

exports[`render() correct with value=true 1`] = `
<div
  className="root"
>
  <RadioGroup
    disabled={false}
    onChange={[Function]}
    size="large"
    value="true"
  >
    <RadioButton
      prefixCls="ant-radio-button"
      value="true"
    >
      Yes
    </RadioButton>
    <RadioButton
      prefixCls="ant-radio-button"
      value="false"
    >
      No
    </RadioButton>
  </RadioGroup>
</div>
`;
